### PR TITLE
updates to /rfqm/v2 endpoints

### PR DESCRIPTION
**Pull request by phil-ociraptor**

### CODE REVIEW

File: src/types.ts [hunk 1]
```diff
@@ -137,8 +142,8 @@ export type FirmQuoteResponse = VersionedQuote<'3', V3RFQFirmQuote> | VersionedQ
 // Implement quoter that is version agnostic
 export interface Quoter {
     fetchIndicativeQuoteAsync(takerRequest: TakerRequest): Promise<IndicativeQuoteResponse>;
+    fetchIndicativeOtcQuoteAsync(takerRequest: TakerRequest): Promise<IndicativeOtcQuote>;
     fetchFirmQuoteAsync(takerRequest: TakerRequest): Promise<FirmQuoteResponse>;
-    fetchFirmOtcQuoteAsync(takerRequest: TakerRequest): Promise<OtcOrderFirmQuoteResponse>;
```
Comment (PirosB3) [inline src/types.ts:141]: YASS:)

**Review by PirosB3 — COMMENTED** [review]
Do we need to bump the major package number?

File: src/handlers.ts [hunk 2]
```diff
@@ -30,6 +30,19 @@ export const generateApiKeyHandler = (): express.RequestHandler => {
...
 };
 
+export const fetchOtcPriceHandler = async (quoter: Quoter, req: express.Request, res: express.Response) => {
+    const takerRequestResponse = parseTakerRequest(req);
+
+    if (!takerRequestResponse.isValid) {
+        return res.status(HttpStatus.BAD_REQUEST).json({ errors: takerRequestResponse.errors });
+    }
+
+    const response = await quoter.fetchIndicativeOtcQuoteAsync(takerRequestResponse.takerRequest);
```
Comment (PirosB3) [inline src/handlers.ts:40]: do we need a try/catch around this?

File: src/request_parser.ts [hunk 3]
```diff
@@ -47,6 +46,11 @@ export const parseTakerRequest = (req: Pick<express.Request, 'headers' | 'query'
             apiKey = undefined;
         }
 
+        let requestUuid = req.headers[ZERO_EX_REQUEST_UUID_HEADER_STRING];
+        if (typeof requestUuid !== 'string') {
+            requestUuid = undefined;
```
Comment (PirosB3) [inline src/request_parser.ts:51]: Shouldn't we be enforcing this?

**Review by PirosB3 — APPROVED** [review]

File: src/request_parser.ts (hunk 3 above)
Comment (phil-ociraptor) [reply src/request_parser.ts:51]: I'm reusing this parser across both `/price` and `/rfqm/v2/price` so it's optional unfortunately

File: src/handlers.ts (hunk 2 above)
Comment (phil-ociraptor) [reply src/handlers.ts:40]: express will handle this, so should be ok - however, I just found something that I missed

**Comment by phil-ociraptor** [comment]
Yes, major version bump coming!

**Pull request merged**